oak_containers_orchestrator/src/container_runtime.rs

```diff
@@ -131,16 +132,23 @@ pub async fn run(container_bundle: &[u8]) -> Result<(), anyhow::Error> {
     };
     rmount_dir(std::path::PathBuf::from("/proc"), container_proc_path).await?;
 
-    // Start the trusted application in a chroot jail of the container.
+    // Change the orchestrator's root to the newly created container rootfs.
+    // This will ensure that the newly launched process will also exist in
+    // this chroot jail.
+    // Note though that following this command, the orchestrator will no longer
+    // be able to access files outside of the chroot jail. This is not a problem
+    // given the orchestrators current feature set, but may become one later.
+    std::os::unix::fs::chroot(container_rootfs_path)?;
+
+    // Change to the specifed working directory.
+    std::env::set_current_dir(oci_filesystem_bundle_config.process.cwd)?;
+
+    // Start the trusted application.
     run_command_and_log_output(
-        tokio::process::Command::new("chroot")
-            .arg(format!(
-                "--userspec={}:{}",
-                oci_filesystem_bundle_config.process.user.uid,
-                oci_filesystem_bundle_config.process.user.gid,
-            ))
-            .arg(container_rootfs_path)
-            .args(oci_filesystem_bundle_config.process.args),
+        tokio::process::Command::new(oci_filesystem_bundle_config.process.args[0].clone())
```

Comment:
I think we can have our cake and eat it too.

Use `tokio::process::Command::pre_exec` to run a closure that has the `chroot` and `set_current_dir` in it. This way, the spawned process will run in the chroot jail, while the orchestrator itself stays outside of it.

Reply:
Fancy! Nice find, am a big fan, was looking for a suitable rust implementation of fork. This fits the bill.